Realm/Realm/Configurations/SyncConfiguration.cs

```diff
@@ -125,17 +128,17 @@ public SyncConfiguration(ObjectId? partition, User user, string optionalPath = n
         /// </param>
         [SuppressMessage("Design", "CA1062:Validate arguments of public methods", Justification = "Arguments are validated in the private ctor.")]
         public SyncConfiguration(Guid? partition, User user, string optionalPath = null)
-            : this((object)partition, user, optionalPath)
+            : this(user, partition, optionalPath)
         {
         }
 
-        private SyncConfiguration(object partition, User user, string path)
+        private SyncConfiguration(User user, RealmValue partition, string path)
```

Comment:
The change in ordering was done so that we could use the implicit conversion operators from the other constructors.